packages/itwinui-react/src/core/Select/Select.types-test.tsx

```diff
@@ -7,24 +7,58 @@
  * (e.g. react-workshop, unit tests, etc.)
  */
 
-import React from 'react';
+import React, { useRef } from 'react';
 import Select from './Select.js';
 
-const App = () => {
+() => {
+  const ref = useRef(null);
   return (
     <>
-      <Select<string>
+      <Select
         options={[
-          { value: '1', label: '1' },
-          { value: '2', label: '2' },
-          { value: '3', label: '3' },
+          { value: 1, label: 'Option 1' },
+          { value: 2, label: 'Option 2' },
+          { value: 3, label: 'Option 3' },
         ]}
         onChange={(value) => {
-          console.log(value);
+          return value;
```

Comment:
To confirm whether value indeed has the correct type, you can use the `as` keyword.

E.g. if we want value to be a `number` in this case, you can have something like:

```jsx
const returnValue: number = value;
return returnValue;
```

That way, if value is not a number, an error would be given.

Reply:
is this the right way?

i was thinking something like https://vitest.dev/guide/testing-types.html

Reply:
I think `.types-test.tsx` files are treated as regular tsx files that users would write. And the type checks are happening when running `tsc` and not `jest`.

So, I don't think using any `jest` function would work. Maybe as `assert typeof value === "number"`? Maybe some of the things @siddhantrawal can try.

Reply:
it's not a jest function (that link is from vitest), we would have to implement our own wrappers and then we can use them anywhere

or maybe it's time to migrate to vitest (part of #1571)

Reply:
Ohh, since we haven't migrated to vitest, by that vitest link, I thought you meant a similar function from jest.

If we want to test types with `vitest` and not using `tsc`, then I think we also need to make changes to the `test` script in `package.json` to run vitest on this file. If so, this may be a good starting point to migrate to vitest.

Reply:
so two tasks for @siddhantrawal:
1. in this PR, use this:
   ```suggestion
             const returnValue: number = value;
             return returnValue;
   ```
2. in a subsequent PR, migrate from `jest` and types tests to `vitest`